### PR TITLE
EOS-26289 : Multipart API creation

### PR DESCRIPTION
**Pull request by priyanka25081999**
# Problem Statement
- Multipart API creation : Create, upload and complete
- Add class and tests files
- Update Makefile
- Implement combine test file for all 3 API
- Experiment the api and added into 'Experiments' folder

Signed-off-by: Priyanka Salunke <priyanka.s.salunke@seagate.com>

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by priyanka25081999** [comment]
Jenkins build passed : http://eos-jenkins.colo.seagate.com/job/Multisite/job/Run-Tests/19/

File: s3/replication/common/experiments/s3_upload_part.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+async def main():
+    async with aiohttp.ClientSession() as session:
+        config = Config()
+
+        bucket_name = config.source_bucket_name
+        object_name = config.object_name_prefix
+        total_parts = int(config.total_parts)
+        object_size = int(config.object_size)
+
+        # Upload id
```
Comment (rajkumarpatel2602) [inline s3/replication/common/experiments/s3_upload_part.py:45]: # Provide actual upload ID in place of __UPLOAD_ID__

File: s3/replication/common/src/s3replicationcommon/s3_complete_multipart_upload.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        return self._final_etag
+
+    async def complete_upload(self):
+        request_uri = AWSV4Signer.fmt_s3_request_uri(
+            self._bucket_name, self._object_name)
+        query_params = urllib.parse.urlencode({'uploadId': self._upload_id})
+        body = ""
+
+        # Prepare xml format
+        str1 = "<CompleteMultipartUpload>"
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_complete_multipart_upload.py:79]: s/str1/tag_str/g

File: s3/replication/common/src/s3replicationcommon/s3_upload_part.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        # XXX Take real pause action
+
+    def resume(self):
+        self._state = S3RequestState.PAUSED
+        # XXX Take real resume action
+
+    def abort(self):
+        self._state = S3RequestState.ABORTED
+        # Abort the reader so that PUT can stop.
+        self._data_reader.abort()
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_upload_part.py:157]: Can be removed as not being used or contributing to actual funcationality.

File: s3/replication/common/tests/system/config/config.yaml [hunk 4]
```diff
@@ -9,9 +9,10 @@ object_tag_name: user-tag
 object_tag_value: user-tag-value
 range_read_offset: -1
 range_read_length: -1
-object_size: 1024
+object_size: 26214400
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/config/config.yaml:12]: Add comment  : `# i.e. 25MB`

File: s3/replication/common/tests/system/config/config.yaml [hunk 5]
```diff
@@ -9,9 +9,10 @@ object_tag_name: user-tag
...
 total_objects: 100
 transfer_chunk_size: 4096
+total_parts: 5
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/config/config.yaml:15]: `# Make sure object_size / total_parts >= 5Mb for all parts.`

File: s3/replication/common/tests/system/config/config.yaml [hunk 6]
```diff
@@ -27,9 +28,10 @@ iam_role: <user_account_number> # User's Account ID
 #object_tag_value: user-tag-value
 #range_read_offset: -1
 #range_read_length: -1
-#object_size: 1024
+#object_size: 26214400
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/config/config.yaml:31]: same as above

File: s3/replication/common/src/s3replicationcommon/s3_complete_multipart_upload.py (hunk 2 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_complete_multipart_upload.py]: done

File: s3/replication/common/src/s3replicationcommon/s3_upload_part.py (hunk 3 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_upload_part.py]: sure. done

File: s3/replication/common/tests/system/config/config.yaml (hunk 4 above)
Comment (priyanka25081999) [reply s3/replication/common/tests/system/config/config.yaml]: done

File: s3/replication/common/tests/system/config/config.yaml (hunk 5 above)
Comment (priyanka25081999) [reply s3/replication/common/tests/system/config/config.yaml]: done

File: s3/replication/common/tests/system/config/config.yaml (hunk 6 above)
Comment (priyanka25081999) [reply s3/replication/common/tests/system/config/config.yaml]: comments added. done

**Review by rajkumarpatel2602 — APPROVED** [review]

**Review by mbcortx — APPROVED** [review]

**Pull request merged**